app/src/cpp/cfc-cpp/MultiThreadedDecoder.h

```diff
@@ -121,9 +136,35 @@ inline void MultiThreadedDecoder::stop()
 	_pool.stop();
 }
 
-inline bool MultiThreadedDecoder::legacy_mode() const
+unsigned MultiThreadedDecoder::fountain_chunk_size(int mode_val)
+{
+	return cimbar::Config::fountain_chunk_size(cimbar::Config::ecc_bytes(), cimbar::Config::symbol_bits() + cimbar::Config::color_bits(), mode_val==4);
+}
+
+inline int MultiThreadedDecoder::mode() const
+{
+	return _modeVal;
+}
+
+inline bool MultiThreadedDecoder::set_mode(int mode_val)
+{
+	if (_modeVal == mode_val)
+		return true;
+
+	if (mode_val != 0 and _writer.chunk_size() != fountain_chunk_size(mode_val))
+		return false; // if so, we need to reset to change it
```

Comment:
In the past (for example when from 4-color to 8-color, in 0.5.x), we had to throw away the MultiThreadedDecoder object and get a new one any time we changed anything. Since we have an "autodetect mode" now, we want to only do that when we must (since it currently requires throwing away progress).

The fountain_chunk_size being different is the condition. Which is to say, *if* the modes had the same fountain_chunk_size, we would be able to make continuous progress on the stream no matter what mode we were in. But there are other things that might prevent that, and in any case there are good reasons `mode B` has to use a different number.